roles/kubernetes/master/tasks/kubeadm-setup.yml

```diff
@@ -29,7 +29,54 @@
     dest: "{{ kube_config_dir }}/kubeadm-config.yaml"
   register: kubeadm_config
 
-- name: kubeadm | Initialize cluster
+- name: Check if kubeadm has already run
+  stat:
+    path: "{{ kube_config_dir }}/admin.conf"
+  register: admin_conf
+
+
+- name: kubeadm | Initialize first master
+  command: timeout -k 240s 240s kubeadm init --config={{ kube_config_dir }}/kubeadm-config.yaml --skip-preflight-checks
+  register: kubeadm_init
+  #Retry is because upload config sometimes fails
+  retries: 3
+  when: inventory_hostname == groups['kube-master']|first and (kubeadm_config.changed or not admin_conf.stat.exists)
+
+- name: slurp kubeadm certs
+  slurp:
+    src: "{{ item }}"
+  with_items:
```

Comment:
is it worth making this a fact for all the certs?

Reply:
We need to copy them to other masters.

Reply:
We need to copy them to other masters.